src/parser_yang.c

```diff
@@ -4516,6 +4519,61 @@ yang_check_deviate(struct lys_module *module, struct unres_schema *unres, struct
 
         /* ... and replace it with the value specified in deviation */
         dev_target->flags |= deviate->flags & LYS_CONFIG_MASK;
+
+        /* "config" is explicitely set in the node */
+        dev_target->flags |= LYS_CONFIG_SET;
+
+        /* "config" must be set to either "yes" or "no" */
+        assert(dev_target->flags ^ LYS_CONFIG_MASK);
+
+        /* from rfc7950#page-86: All key leafs in a list MUST have the same value
+         * for their "config" as the list itself.
+         */
+        if ((node_list = lys_is_key((const struct lys_node_leaf *)dev_target, NULL))) {
+            if ((node_list->flags & LYS_CONFIG_MASK) != (dev_target->flags & LYS_CONFIG_MASK)) {
+                LOGVAL(module->ctx, LYE_INSTMT, LY_VLOG_NONE, NULL, "config");
+                LOGVAL(module->ctx, LYE_SPEC, LY_VLOG_NONE, NULL,
+                    "The deviate statement causes a violation of \"config\" rules in a key leaf. "
+                    "The \"config\" value in the key leaf must be the same as in the list to which belongs.");
+                goto error;
+            }
+        }
+
+        /* from rfc7950#page-135: If the parent node is a case node,
+         * the value is the same as the case node's parent choice node.
+         */
+        if (dev_target->nodetype == LYS_CASE) {
+            if ((parent = lys_parent(dev_target)) && (parent->nodetype == LYS_CHOICE)) {
```

Comment:
I do not think this can ever be false, should probably be an assert and it will save one `if` nesting.